kotlin-checks-test-sources/src/main/kotlin/checks/ArrayHashCodeAndToStringCheckSample.kt

```diff
@@ -0,0 +1,108 @@
+            shortArray.contentToString() +              // Compliant
+            charArray.toString() +                      // Noncompliant {{Use "contentToString()" instead.}}
+            charArray.contentToString() +               // Compliant
+            intArray.toString() +                       // Noncompliant {{Use "contentToString()" instead.}}
+            intArray.contentToString() +                // Compliant
+            longArray.toString() +                      // Noncompliant {{Use "contentToString()" instead.}}
+            longArray.contentToString() +               // Compliant
+            floatArray.toString() +                     // Noncompliant {{Use "contentToString()" instead.}}
+            floatArray.contentToString() +              // Compliant
+            doubleArray.toString() +                    // Noncompliant {{Use "contentToString()" instead.}}
+            doubleArray.contentToString()               // Compliant
+    }
+
+    fun rspecSamples() {
+        var primitiveArray: IntArray = intArrayOf(1, 2, 3)
+        var objectArray: Array<String> = arrayOf("A", "B", "C")
+        var arrayOfArray: Array<Array<String>> = arrayOf(arrayOf("A", "B"), arrayOf("C", "D"))
+
+        println(primitiveArray.toString())       // Noncompliant
+        println(primitiveArray.hashCode())       // Noncompliant
+        println(objectArray.toString())          // Noncompliant
+        println(objectArray.hashCode())          // Noncompliant
+        println(arrayOfArray.toString())         // Noncompliant
+        println(arrayOfArray.contentToString())  // Noncompliant
+        println(arrayOfArray.hashCode())         // Noncompliant
+        println(arrayOfArray.contentHashCode())  // Noncompliant
+
+        println(primitiveArray.contentToString())   // Compliant
+        println(primitiveArray.contentHashCode())   // Compliant
+        println(objectArray.contentDeepToString())  // Compliant
+        println(objectArray.contentDeepHashCode())  // Compliant
+        println(arrayOfArray.contentDeepToString()) // Compliant
+        println(arrayOfArray.contentDeepHashCode()) // Compliant
+    }
+
+    fun falseNegative() {
+        val arrayOfArray: Array<Array<String>> = arrayOf(arrayOf("A", "B"), arrayOf("C", "D"))
+        // contrary to the above "rspecSamples()" version of "arrayOfArray", here the declaration use "val" instead of "var"
+        // and the "predictReceiverExpression(...)" is using the type the initializer "T" coming from " arrayOf(vararg elements: T)"
+        // instead of the explicit "Array<Array<String>>" variable type, which is more specific in this case.
+        // Furthermore, resolving as "T" is a mistake, is should be at least "Array<T>", because of the "vararg" modifier.
+        println(arrayOfArray.contentToString())  // false-negative
+        println(arrayOfArray.contentHashCode())  // false-negative
+    }
```

Comment:
Good catch! This is actually a small bug in `determineType`. If you simply replace the line 
```
            is KtCallExpression -> getResolvedCall(bindingContext)?.getReturnType()
```
with
```
            is KtCallExpression -> getResolvedCall(bindingContext)?.resultingDescriptor?.returnType
```
it will work ([ApiExtensions.kt:485](https://github.com/SonarSource/sonar-kotlin/blob/513ab31ec6514a24621bad242b3195c00b3f16e4/sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/ApiExtensions.kt#L485))! :)